### PR TITLE
Fix several bugs in relative datetime filters

### LINKED ISSUE #22222: In the relative date picker, changing the granularity of the Past/Next range picker should change the granularity of the Starting From picker
When I change the time granularity of the Past/Next option, e.g. from Days to Months, the granularity of the Starting From option doesn’t change to match, but is supposed to. 

The reverse should not be true; i.e. changing the granularity of the “Starting from” option should not change the granularity of the Past/Next picker, since I might want to do something like the past 30 days starting 2 years ago)

image

### LINKED ISSUE #22224: `Include this quarter` action is missing the words "this quarter"
When the granularity for a relative filter is set to `Quarter`, the include-current-period action currently just says "Include." It should read `Include this quarter`.

image

This has been a problem since version 0.42 at least:

image

### LINKED ISSUE #22225: Date preview is wrong when using the "Starting from…" action when it pushes back before the current year
The date range preview is wrong if your "Starting from…" option pushes your range back before 2022, only showing a 1-day "range" like `Apr 28, 2021 - Apr 28, 2021`

image

image

### LINKED ISSUE #22227: Can't change a relative month filter's values when using the "starting from" feature
To repro:
- Open up the Orders table
- Click on the Created At column and click the "filter on this column action," then select relative dates
- Set the filter to previous 1 month, starting from 3 months ago, and add the filter
- Click on the filter token in the top-left to reopen the filter
- Try changing the 1 month to any other quantity, like 3 months, then try to update the filter
- The changes are not applied, and when you reopen the filter again it has reverted to the previous value

https://user-images.githubusercontent.com/1447303/166149987-33e3248e-a1b1-4ac6-9bed-deebdea5ff81.mp4

https://www.loom.com/share/f43cf9608f3a47af91daa9b8fcb8b14a

### LINKED ISSUE #22228: "Next" relative filter tokens should read "from now" instead of "ago"
To repro:
- Open up Orders, click on the Created At field, filter on this column, then relative dates
- Select the **Next** tab, then click the ellipses and select the starting-from action
- Add the filter
- Notice that it reads something like "Created at Next 30 days, starting from 7 days **ago**" — it should actually read "Created at Next 30 days, starting from 7 days **from now**"

image

### PR DESCRIPTION
**Pull request by akiselev**
**Changing the granularity of the `Past`/`Next` range picker now changes the granularity of the `Starting from` option**
  - Fixes #22222
  - Reproduces #22222 

---

**`Include this quarter` now displays correctly**
  - Fixes #22224 

---

**Now shows the correct date preview in `Past`/`Next` range picker**
  - Fixes #22225 
  - Reproduces #22225 

---

**Can now change `Past`/`Next` granularity when the value is set to `1`**
  - Fixes #22227
  - Reproduces #22227

---

**`Next` relative filter pill text now reads `from now` instead of `ago`**
  - Fixes #22228
  - Reproduces #22228

### CODE REVIEW

File: frontend/src/metabase/lib/query_time.js [hunk 1]
```diff
@@ -273,6 +279,8 @@ export function parseFieldBucketing(field, defaultUnit = null) {
   const dimension = FieldDimension.parseMBQLOrWarn(field);
   if (dimension) {
     return dimension.temporalUnit() || defaultUnit;
+  } else if (field?.[0] === "+" && field?.[1]?.[0] === "field") {
```
Comment (alxnddr) [inline frontend/src/metabase/lib/query_time.js:282]: It is quite hard to understand, can you please rewrite this condition in a [self-descriptive way](https://www.notion.so/metabase/FE-Style-Guide-Proposals-Discussion-acdddeb36bd148e592d8be749ff9e4ef#f9be258edecb4896bac15cc5d15ad871)?

File: frontend/test/metabase/scenarios/filters/relative-datetime.cy.spec.js [hunk 2]
```diff
@@ -158,6 +158,113 @@ describe("scenarios > question > relative-datetime", () => {
...
     });
+
+    it("should change the starting from units to match (metabase#22222)", () => {
+      openOrdersTable();
+
+      openCreatedAt("Past");
+      addStartingFrom();
+      popover().within(() => cy.findByText("days").click());
+      popover()
+        .last()
```
Comment (alxnddr) [inline frontend/test/metabase/scenarios/filters/relative-datetime.cy.spec.js:169]: Using `data-testid` for cases with multiple popovers would be more understandable here and below

**Review by alxnddr — APPROVED** [review]
Works as expected, please take a look at comments I left

File: frontend/test/metabase/scenarios/filters/relative-datetime.cy.spec.js (hunk 2 above)
Comment (akiselev) [reply frontend/test/metabase/scenarios/filters/relative-datetime.cy.spec.js]: I couldn't add test IDs to the `Select` component cleanly so I separated the logic out into functions with more descriptive names

File: frontend/src/metabase/lib/query_time.js (hunk 1 above)
Comment (akiselev) [reply frontend/src/metabase/lib/query_time.js]: Fixed

**Review by mazameli — APPROVED** [review]
Nice!

**Pull request merged**